sdk/src/runtime/lib/db/DOWorkerDialect.ts

```diff
@@ -92,13 +100,68 @@ class DOWorkerConnection implements DatabaseConnection {
   }): Promise<QueryResult<R>> {
     log("Forwarding query to Durable Object: %s", compiledQuery.sql);
 
+    const parameters = this.processParameters(compiledQuery.parameters);
+
     // Call the DO's kyselyExecuteQuery method
     const result = await this._executeQuery({
       sql: compiledQuery.sql,
-      parameters: compiledQuery.parameters,
+      parameters,
     });
 
-    return result as QueryResult<R>;
+    const rows = this.processRows(result.rows);
+
+    return { ...result, rows } as QueryResult<R>;
+  }
+
+  processParameters(parameters: readonly unknown[]) {
+    if (!this.typeConverters) {
+      return parameters;
+    }
+
+    return parameters.map((parameter) => {
+      if (parameter === null || parameter === undefined) {
+        return parameter;
+      }
+
+      const constructorName = (parameter as any).constructor?.name;
+      if (
+        constructorName &&
+        this.typeConverters![constructorName]?.serialize
+      ) {
+        return this.typeConverters![constructorName].serialize!(parameter);
+      }
+
+      return parameter;
+    });
+  }
+
+  processRows(rows: any[]) {
+    if (!this.typeConverters || !rows || rows.length === 0) {
+      return rows;
+    }
+
+    return rows.map((row) => {
+      const newRow = { ...row };
+      for (const key in newRow) {
+        const value = newRow[key];
+        // 1. Check for exact column name match
+        if (this.typeConverters![key]?.parse) {
+          newRow[key] = this.typeConverters![key].parse!(value, key);
+          continue;
+        }
```

Comment:
I think it might be a bit too implicit that exact column names also count as matches, or might come as a surprise to some users.

What do you think of us relying on `match()` always? That way its explicit what matches and what doesn't.